Make LayerRepaintRects internal to RenderLayer

smfr opened the pull request:
#### d24752c38292065d9216c2adbd11a6f60f2b750c
<pre>
Make LayerRepaintRects internal to RenderLayer
<a href="https://bugs.webkit.org/show_bug.cgi?id=264530">https://bugs.webkit.org/show_bug.cgi?id=264530</a>
<a href="https://rdar.apple.com/118203951">rdar://118203951</a>

Reviewed by Chris Dumez.

There are two call sites that make use of RenderLayer&apos;s cached repaint rects, but
they both use only the clippedOverflowRect, not the outlineBoundsRect, so we can
hide RepaintRects inside of RenderLayer, and only expose the cachedClippedOverflowRect()
to those two clients.

* Source/WebCore/rendering/LegacyLineLayout.cpp:
(WebCore::LegacyLineLayout::layoutRunsAndFloats):
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::RenderLayer::cachedClippedOverflowRect const):
(WebCore::RenderLayer::setRepaintRects):
(WebCore::RenderLayer::shouldRepaintAfterLayout const):
* Source/WebCore/rendering/RenderLayer.h:
(WebCore::RenderLayer::repaintRects const):
* Source/WebCore/rendering/RenderLayerModelObject.cpp:
(WebCore::RenderLayerModelObject::styleDidChange):
(WebCore::RenderLayerModelObject::cachedLayerClippedOverflowRect const):
(WebCore::RenderLayerModelObject::layerRepaintRects const): Deleted.
* Source/WebCore/rendering/RenderLayerModelObject.h:
* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
(WebCore::RenderLayerScrollableArea::scrollTo):

Canonical link: <a href="https://commits.webkit.org/270550@main">https://commits.webkit.org/270550@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c33e0528120109cb4825f04389a3326f898d0688

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/25644 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/4249 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/26927 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/27743 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/23490 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/25927 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/5996 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/1684 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/23621 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/25893 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3162 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22101 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/28324 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/2794 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/23053 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29140 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/23399 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/23419 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/26995 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/2814 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1051 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4189 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6188 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/3258 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3130 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/rendering/RenderLayer.h:
```diff
@@ -969,10 +963,24 @@ class RenderLayer : public CanMakeWeakPtr<RenderLayer>, public CanMakeCheckedPtr
+
+    std::optional<RepaintRects> repaintRects() const
+    {
+        if (m_repaintRectsValid)
```
nit, not super consistent with RenderLayer::cachedClippedOverflowRect() which you did the opposite and used std::nullopt.

smfr committed:
Make LayerRepaintRects internal to RenderLayer
https://bugs.webkit.org/show_bug.cgi?id=264530
rdar://118203951

Reviewed by Chris Dumez.

There are two call sites that make use of RenderLayer's cached repaint rects, but
they both use only the clippedOverflowRect, not the outlineBoundsRect, so we can
hide RepaintRects inside of RenderLayer, and only expose the cachedClippedOverflowRect()
to those two clients.

* Source/WebCore/rendering/LegacyLineLayout.cpp:
(WebCore::LegacyLineLayout::layoutRunsAndFloats):
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::RenderLayer::cachedClippedOverflowRect const):
(WebCore::RenderLayer::setRepaintRects):
(WebCore::RenderLayer::shouldRepaintAfterLayout const):
* Source/WebCore/rendering/RenderLayer.h:
(WebCore::RenderLayer::repaintRects const):
* Source/WebCore/rendering/RenderLayerModelObject.cpp:
(WebCore::RenderLayerModelObject::styleDidChange):
(WebCore::RenderLayerModelObject::cachedLayerClippedOverflowRect const):
(WebCore::RenderLayerModelObject::layerRepaintRects const): Deleted.
* Source/WebCore/rendering/RenderLayerModelObject.h:
* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
(WebCore::RenderLayerScrollableArea::scrollTo):

Canonical link: https://commits.webkit.org/270550@main
